designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/scenarioActivity/Endpoints.scala

```diff
@@ -0,0 +1,216 @@
+import pl.touk.nussknacker.ui.server.HeadersSupport.FileName
+import pl.touk.nussknacker.ui.server.TapirStreamEndpointProvider
+import sttp.model.HeaderNames
+import sttp.model.StatusCode.{InternalServerError, NotFound, Ok}
+import sttp.tapir._
+import sttp.tapir.docs.openapi.OpenAPIDocsOptions
+import sttp.tapir.json.circe.jsonBody
+import sttp.tapir.server.ServerEndpoint
+import sttp.tapir.swagger.SwaggerUIOptions
+import sttp.tapir.swagger.bundle.SwaggerInterpreter
+
+import scala.concurrent.Future
+
+object Endpoints extends BaseEndpointDefinitions {
+
+  import TapirCodecs.ContentDispositionCodec._
+  import TapirCodecs.HeaderCodec._
+  import TapirCodecs.ScenarioNameCodec._
+  import TapirCodecs.VersionIdCodec._
+  import pl.touk.nussknacker.ui.api.description.scenarioActivity.Dtos.ScenarioActivityError._
+  import pl.touk.nussknacker.ui.api.description.scenarioActivity.Dtos._
+  import pl.touk.nussknacker.ui.api.description.scenarioActivity.InputOutput._
+
+  val apiDocumentTitle = "Nussknacker Scenario Activity API"
+
+  val apiVersion = "1.1.0"
+
+  def swaggerEndpoints(implicit streamProvider: TapirStreamEndpointProvider): List[ServerEndpoint[Any, Future]] =
+    SwaggerInterpreter(
+      swaggerUIOptions = SwaggerUIOptions.default.copy(
+        pathPrefix = "api" :: "processes" :: "docs" :: Nil,
+      ),
+      openAPIInterpreterOptions = openAPIOptions
+    ).fromEndpoints(apiEndpoints, apiDocumentTitle, apiVersion)
+
+  def openAPIOptions: OpenAPIDocsOptions =
+    OpenAPIDocsOptions.default.copy(markOptionsAsNullable = true)
+
+  def apiEndpoints(implicit streamProvider: TapirStreamEndpointProvider): List[PublicEndpoint[_, _, _, Any]] = List(
+    scenarioActivityEndpoint,
+    scenarioActivitiesCountEndpoint,
+    scenarioActivitiesSearchEndpoint,
+    scenarioActivitiesMetadataEndpoint,
+    scenarioActivitiesEndpoint,
+    addCommentEndpoint,
+    editCommentEndpoint,
+    deleteCommentEndpoint,
+    attachmentsEndpoint,
+    addAttachmentEndpoint,
+    downloadAttachmentEndpoint,
+  )
+
+  lazy val scenarioActivityEndpoint
+      : PublicEndpoint[ProcessName, ScenarioActivityError, ScenarioCommentsAndAttachments, Any] =
+    baseNuApiEndpoint
+      .summary("Deprecated scenario comments and attachments service")
+      .tag("Activities")
+      .get
+      .in("processes" / path[ProcessName]("scenarioName") / "activity")
+      .out(
+        statusCode(Ok).and(jsonBody[ScenarioCommentsAndAttachments].example(Examples.scenarioCommentsAndAttachments))
+      )
+      .errorOut(scenarioNotFoundErrorOutput)
+      .deprecated()
+
+  lazy val scenarioActivitiesMetadataEndpoint
+      : PublicEndpoint[ProcessName, ScenarioActivityError, ScenarioActivitiesMetadata, Any] =
+    baseNuApiEndpoint
+      .summary("Scenario activities metadata service")
+      .tag("Activities")
+      .get
+      .in("processes" / path[ProcessName]("scenarioName") / "activity" / "activities" / "metadata")
```

Comment:
With this part we want to show separation from an old and a new API?

Reply:
After discussion with @Dzuming we decided to to filtering/search/pagination on FE at this stage. It may be necessary to add it later on BE because of performance issues, but not at this moment.